v3/integrations/nrredis-v9/nrredis.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package nrredis instruments github.com/redis/go-redis/v9.
+//
+// Use this package to instrument your redis/go-redis/v9 calls without having to
+// manually create DatastoreSegments.
+package nrredis
+
+import (
+	"context"
+	"net"
+	"strings"
+
+	"github.com/newrelic/go-agent/v3/internal"
+	newrelic "github.com/newrelic/go-agent/v3/newrelic"
+	redis "github.com/redis/go-redis/v9"
+)
+
+func init() { internal.TrackUsage("integration", "datastore", "redis") }
+
+type contextKeyType struct{}
+
+type hook struct {
+	segment newrelic.DatastoreSegment
+}
+
+var _ redis.Hook = (*hook)(nil)
+
+var (
+	segmentContextKey = contextKeyType(struct{}{})
+)
+
+// NewHook creates a redis.Hook to instrument Redis calls.  Add it to your
+// client, then ensure that all calls contain a context which includes the
+// transaction.  The options are optional.  Provide them to get instance metrics
+// broken out by host and port.  The hook returned can be used with
+// redis.Client, redis.ClusterClient, and redis.Ring.
+func NewHook(opts *redis.Options) redis.Hook {
+	h := hook{}
+	h.segment.Product = newrelic.DatastoreRedis
+	if opts == nil {
+		return h
+	}
+
+	// Per https://pkg.go.dev/github.com/redis/go-redis#Options the
+	// network should either be tcp or unix, and the default is tcp.
+	if opts.Network == "unix" {
+		h.segment.Host = "localhost"
+		h.segment.PortPathOrID = opts.Addr
+		return h
+	}
+	if host, port, err := net.SplitHostPort(opts.Addr); err == nil {
+		if host != "" {
```

Comment:
```suggestion
		if host == "" {
```
My bad, this should have been an equality check.
I've tested the edit on my local go 1.20.2 and the test should work.